### PR TITLE
Add RPM and image builds

### PR DESCRIPTION
**Pull request by spadgett**
First cut at setting up RPM and image builds using the instructions here:

https://github.com/openshift/release/tree/master/tools/hack/golang

Probably conflicts with #5 

See also https://github.com/openshift/image-registry/pull/6

cc @jwforres @stevekuznetsov

### CODE REVIEW

**Comment by spadgett** [comment]
I haven't gotten a build to successfully run yet. @stevekuznetsov I will probably need some help with this.

First error I'm hitting:

```
$ make 
hack/build-go.sh
/Users/sam/go/src/github.com/openshift/origin-web-console-server/hack/lib/constants.sh: line 59: OS_GIT_MAJOR: unbound variable
[ERROR] PID 24363: hack/lib/constants.sh:42: `ldflags+=($(os::build::ldflag "${OS_GO_PACKAGE}/pkg/version.majorFromGit" "${OS_GIT_MAJOR}"))` exited with status 1.
[INFO] 		Stack Trace:
[INFO] 		  1: hack/lib/constants.sh:42: `ldflags+=($(os::build::ldflag "${OS_GO_PACKAGE}/pkg/version.majorFromGit" "${OS_GIT_MAJOR}"))`
[INFO] 		  2: hack/lib/build/binaries.sh:177: os::build::ldflags
[INFO] 		  3: hack/lib/build/binaries.sh:161: os::build::internal::build_binaries
[INFO] 		  4: hack/build-go.sh:20: os::build::build_binaries
[INFO]   Exiting with code 1.
[ERROR] PID 24363: hack/lib/build/binaries.sh:171: `version_ldflags=$(os::build::ldflags)` exited with status 1.
[INFO] 		Stack Trace:
[INFO] 		  1: hack/lib/build/binaries.sh:171: `version_ldflags=$(os::build::ldflags)`
[INFO] 		  2: hack/lib/build/binaries.sh:161: os::build::internal::build_binaries
[INFO] 		  3: hack/build-go.sh:20: os::build::build_binaries
[INFO]   Exiting with code 1.
[ERROR] hack/build-go.sh exited with code 1 after 00h 00m 01s
make: *** [all] Error 1
```

Looks like I'm also picking up additional files that weren't in openshift/image-registry#6

**Comment by spadgett** [comment]
openshift-ci is already laughing at me 👍

**Comment by spadgett** [comment]
Looks like this is failing because we have no tags.

**Comment by stevekuznetsov** [comment]
@spadgett what is you timeline on this? let's jump on a partner-program session and hash this out this week?

**Comment by spadgett** [comment]
@stevekuznetsov Thanks, really appreciate it. Timeline is 3.9, so we have some time. I'll be here tomorrow, but I'm on PTO next week.

**Comment by stevekuznetsov** [comment]
@spadgett ok -- let's pick up tomorrow and work on whatever you have left

**Comment by spadgett** [comment]
Success!

```
[openshift/origin-web-console-server] --> Tagged as openshift/origin-web-console-server:latest
[openshift/origin-web-console-server] --> Done
[INFO] hack/build-images.sh exited with code 0 after 00h 03m 35s
```

**Comment by spadgett** [comment]
@stevekuznetsov I have things working, but a couple questions:

1. Should we just create a dummy / pre-release tag to avoid the `﻿﻿OS_GIT_MAJOR: unbound variable` error? `make` fails for me without since there are no tags in this repository
2. Is there a way to build the image on a Mac? I had to use a Fedora VM.

**Comment by spadgett** [comment]
@jwforres I split out my changes into a separate commit for review and added the vendor console script. Tested vendor, and it works with no other changes.

File: origin-web-console-server.spec [hunk 1]
```diff
@@ -138,5 +138,5 @@ done
 %pre
 
 %changelog
-* Mon Nov 06 2017 Anonymous <anon@nowhere.com> 0.0.1
-- Initial example of spec.
+* Thu Nov 16 2017 Samuel Padgett <spadgett@redat.com> 0.0.1
```
Comment (jwforres) [inline origin-web-console-server.spec]: typo on your email address :)   redhat

**Comment by jwforres** [comment]
generally LGTM, just need to get the OS_GIT_MAJOR issue dealt with

**Comment by spadgett** [comment]
Thanks, fixed the typo.

There's still a description to add here. Any suggestions?

```
Name:           %{package_name}
Version:        %{version}
Release:        %{release}%{?dist}
Summary:        TODO
License:        ASL 2.0
URL:            https://%{import_path}
```

edit: made it this for now

```
Summary:        Web Console for the OpenShift Application Platform
```

**Comment by stevekuznetsov** [comment]
Why not `v3.9.0-alpha.0` or whatever? Do we not expect this repo to have tags?

**Comment by spadgett** [comment]
Created a `v3.9.0-alpha.0` tag. Renamed image to `origin-web-console`. Travis is green.

@jwforres PTAL

**Comment by spadgett** [comment]
@jupierce Just to close the loop here: we have RPM builds working for origin-web-console. (I don't know Adam's GitHub ID to tag him.)

**Comment by jwforres** [comment]
all LGTM, you can merge when ready

**Comment by jupierce** [comment]
@spadgett I'm working on capturing the process for getting your RPMs and images built here: https://trello.com/c/1aQdAGZN if you want to go ahead and create a card from this template.

**Pull request merged**